discovery/kubernetes/endpoints.go

```diff
@@ -152,31 +110,102 @@ func (e *Endpoints) Run(ctx context.Context, ch chan<- []*targetgroup.Group) {
 		},
 	})
 
+	return e
+}
+
+func (e *Endpoints) enqueue(obj interface{}) {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+	if err != nil {
+		return
+	}
+
+	e.queue.Add(key)
+}
+
+// Run implements the Discoverer interface.
+func (e *Endpoints) Run(ctx context.Context, ch chan<- []*targetgroup.Group) {
+	defer e.queue.ShutDown()
+
+	cacheSyncs := []cache.InformerSynced{
+		e.endpointsInf.HasSynced,
+		e.serviceInf.HasSynced,
+		e.podInf.HasSynced,
+	}
+	if !cache.WaitForCacheSync(ctx.Done(), cacheSyncs...) {
```

Comment:
as the method is written with a variadic parameter so let's use it that way, then there's no need for the []cache.InformerSynced slice before.

Reply:
Done in https://github.com/prometheus/prometheus/commit/e2a3a3f6260a816a0b77fab6c191cd685db2a973.